src/tools/compiletest/src/runtest.rs

```diff
@@ -1242,6 +1242,9 @@ impl<'test> TestCx<'test> {
             self.props.from_aux_file(&aux_testpaths.file, self.revision, self.config);
         if aux_type == Some(AuxType::ProcMacro) {
             aux_props.force_host = true;
+            if aux_props.edition.is_none() {
+                aux_props.edition = Some("2024".to_string());
+            }
```

Comment:
Discussion: I'm generally very much in favor of reducing boiler plate, but I'm hesistant about this particular default:

- Wouldn't defaulting to edition 2024 significantly reduce our test coverage? After all, the compiler needs to still support even edition 2015 and that we can't have new changes regress its behavior on older editions (at least not unintentionally).
- Usually, if some behavior is edition-specific, we *want* to have multiple revisions: e.g. a revision for `//@ edition: 2024` but also an edition for e.g. `//@ edition: 2015`.

I actually consider it a feature that ui tests default to edition 2015 (I think it's 2015?) but have to explicitly revision or opt-in to newer editions if there's edition-specific behavior -- *because* we need to support older editions too.

Reply:
> Wouldn't defaulting to edition 2024 significantly reduce our test coverage?

It wouldn't reduce it would just *shift* it (from 2015 to $LATEST_STABLE / 2024).
We only test a single edition by default, not several. Hence, there's no *reduction*.

We do have several tests that test multiple editions (via revisions) but that's of course not the default.

Reply:
Still, I consider the jump away from 2015 a bit dubious because 2015 and 2018 differ majorly in my eyes because of the module & path resolution changes (which is one point of this PR lol).

So yes, there's a risk that we would no longer exercise the interaction between the pre-2018 modules & paths and proc macros in new tests going forward (which use this directive).

Reply:
> > Wouldn't defaulting to edition 2024 significantly reduce our test coverage?
> 
> It wouldn't reduce it would just _shift_ it (from 2015 to $LATEST_STABLE / 2024). We only test a single edition by default, not several. Hence, there's no _reduction_.

Right, very good point. In that case, we are still *shifting*/*moving* test coverage, so we probably shouldn't shift that.

Reply:
Fair. I personally didn't feel that the test coverage was reduced, since the tests that were explicitly hitting any 2018+ behavior were set with an explicit edition. But I can understand the concerns, so I dropped that commit. We can revisit in the future if someone wants.

Reply:
Yeah, I'm not against shifting the default, as long as we preserve the coverage for older editions too somehow.